include/alpaka/kernel/Traits.hpp

```diff
@@ -215,15 +215,26 @@ namespace alpaka
                 static_assert(std::is_same_v<Result, void>, "The TKernelFnObj is required to return void!");
             }
         };
+
+        // asserts that T is trivially copyable. We put this in a separate function so we can see which T would fail
+        // the test, when called from a fold expression.
+        template<typename T>
+        inline void assertKernelArgIsTriviallyCopyable()
+        {
+            static_assert(
+                std::is_empty_v<T> || std::is_trivially_copyable_v<T>,
```

Comment:
What is `std::is_empty_v<T>` achieving here? Which types are empty but not trivially copyable?

Reply:
The closure types of lambda functions :)